python/tchannel/tornado/peer.py

```diff
@@ -458,35 +492,95 @@ def send(self, arg1, arg2, arg3, traceflag=False, headers=None):
             parent_span_id = None
             trace_id = None
 
-        # set 'as' to 'raw if 'as' is empty
+        # TODO set default transport headers
         headers = headers or {}
-        if 'as' not in headers:
-            headers['as'] = 'raw'
-
-        connection = yield self.peer.connect()
-        message_id = connection.next_message_id()
-        response = yield connection.send_request(
-            Request(
-                service=self.service,
-                argstreams=[InMemStream(endpoint), arg2, arg3],
-                id=message_id,
-                headers=headers,
-                tracing=Trace(
-                    name=endpoint,
-                    trace_id=trace_id,
-                    parent_span_id=parent_span_id,
-                    endpoint=Endpoint(self.peer.host,
-                                      self.peer.port,
-                                      self.service),
-                    traceflags=traceflag,
-                )
+        headers[THeaders.ARG_SCHEME] = headers.get(
+            THeaders.ARG_SCHEME,
+            ArgSchemeFlag.raw,
+        )
+        headers[THeaders.RETRY] = headers.get(
+            THeaders.RETRY,
+            RetryFlag.default,
+        )
+
+        original_request = Request(
+            service=self.service,
+            argstreams=[InMemStream(endpoint), arg2, arg3],
+            id=0,
```

Comment:
This area of code still needs the most work. A lot of `Request` implementation details are leaking into `PeerClientOperation`. A `Request` should know how to send itself and handle retries. The `PCO` should be able to just say `response = yield request.send()` or arguably `response = yield connection.send(request)`.